### ISSUE TITLE
Wrong view predicate name in `view_config` -> less than helpful exception

### ISSUE THREAD

**Issue by ztane** [Pylons/pyramid#1601]
I accidentally typoed `request_method` as `method` in `view_config` and got less-than-helpful error message:

```
Traceback (most recent call last):
  File ".../venv34/bin/pserve", line 9, in <module>
    load_entry_point('pyramid==1.5.1', 'console_scripts', 'pserve')()
  File ".../venv34/lib/python3.4/site-packages/pyramid-1.5.1-py3.4.egg/pyramid/scripts/pserve.py", line 51, in main
    return command.run()
  File ".../venv34/lib/python3.4/site-packages/pyramid-1.5.1-py3.4.egg/pyramid/scripts/pserve.py", line 316, in run
    global_conf=vars)
  File ".../venv34/lib/python3.4/site-packages/pyramid-1.5.1-py3.4.egg/pyramid/scripts/pserve.py", line 340, in loadapp
    return loadapp(app_spec, name=name, relative_to=relative_to, **kw)
  File ".../venv34/lib/python3.4/site-packages/PasteDeploy-1.5.2-py3.4.egg/paste/deploy/loadwsgi.py", line 247, in loadapp
    return loadobj(APP, uri, name=name, **kw)
  File ".../venv34/lib/python3.4/site-packages/PasteDeploy-1.5.2-py3.4.egg/paste/deploy/loadwsgi.py", line 272, in loadobj
    return context.create()
  File ".../venv34/lib/python3.4/site-packages/PasteDeploy-1.5.2-py3.4.egg/paste/deploy/loadwsgi.py", line 710, in create
    return self.object_type.invoke(self)
  File ".../venv34/lib/python3.4/site-packages/PasteDeploy-1.5.2-py3.4.egg/paste/deploy/loadwsgi.py", line 146, in invoke
    return fix_call(context.object, context.global_conf, **context.local_conf)
  File ".../venv34/lib/python3.4/site-packages/PasteDeploy-1.5.2-py3.4.egg/paste/deploy/util.py", line 55, in fix_call
    val = callable(*args, **kw)
  File ".../the-project/project/config/main.py", line 94, in main
    return config.make_wsgi_app()
  File ".../venv34/lib/python3.4/site-packages/pyramid-1.5.1-py3.4.egg/pyramid/config/__init__.py", line 939, in make_wsgi_app
    self.commit()
  File ".../venv34/lib/python3.4/site-packages/pyramid-1.5.1-py3.4.egg/pyramid/config/__init__.py", line 610, in commit
    self.action_state.execute_actions(introspector=self.introspector)
  File ".../venv34/lib/python3.4/site-packages/pyramid-1.5.1-py3.4.egg/pyramid/config/__init__.py", line 1048, in execute_actions
    for action in resolveConflicts(self.actions):
  File ".../venv34/lib/python3.4/site-packages/pyramid-1.5.1-py3.4.egg/pyramid/config/__init__.py", line 1132, in resolveConflicts
    discriminator = undefer(action['discriminator'])
  File ".../venv34/lib/python3.4/site-packages/pyramid-1.5.1-py3.4.egg/pyramid/registry.py", line 250, in undefer
    v = v.resolve()
  File ".../venv34/lib/python3.4/site-packages/pyramid-1.5.1-py3.4.egg/pyramid/registry.py", line 242, in resolve
    return self.func()
  File ".../venv34/lib/python3.4/site-packages/pyramid-1.5.1-py3.4.egg/pyramid/config/views.py", line 1150, in discrim_func
    order, preds, phash = predlist.make(self, **pvals)
  File ".../venv34/lib/python3.4/site-packages/pyramid-1.5.1-py3.4.egg/pyramid/config/util.py", line 165, in make
    % (kw, ','.join(closest)))
pyramid.exceptions.ConfigurationError: Unknown predicate values: {'method': 'POST'}
```

Notably, traceback neither told which method it was nor really gave any hint on how to fix it beyond that I did say something wrong.
- There really should be some hint on how to find the actual view. (like in the case of wrong route_name).
- As for the [exception message](https://github.com/Pylons/pyramid/blob/master/pyramid/config/util.py#L157) could be really made nicer with `difflib.get_close_matches` (POC):
  
  ```
      try:
          from difflib import get_close_matches
          closest = []
          names = [ name for name, _ in ordered ]
          for name in kw:
              closest.extend(get_close_matches(name, names, 3))
  
          raise ConfigurationError(
              'Unknown predicate values: %r (perhaps you meant %s)'
               % (kw, ','.join(closest)))
  
      except ImportError:
          pass
  
      raise ConfigurationError('Unknown predicate values: %r' % (kw,))
  ```

The resulting exception message: `Unknown predicate values: {'method': 'POST'} (perhaps you meant request_method)`. These kind of friendly messages could be used for the wrong route_name and other similar cases too. The exception for wrong route presently is:

```
pyramid.exceptions.ConfigurationExecutionError: <class 'pyramid.exceptions.ConfigurationError'>: No route named lgoout found for view registration
  in:
  Line 288 of file Project/project/views/__init__.py:
    @view_config(route_name="lgoout")
```

**Comment by mmerickel** [comment]
I don't care much about the difflib stuff, that should be a separate issue. However, we should probably fix the output of `unknown predicate values` to point to the appropriate directive/decorator similar to your later example where it points to the decorator on an invalid route name.

**Comment by mmerickel** [comment]
Just a quick note about this. I think this issue is specific to the `add_view` which uses a deferred discriminator. The discriminator is the thing that performs the check, and exceptions occurring there do not map nicely to the action info containing the appropriate linkage back to user code.

**Comment by mmerickel** [comment]
I think this issue is fixed by #3054.
